SimPEG/Regularization.py

```diff
@@ -416,655 +563,1030 @@ def eval2Deriv(self, m, v=None):
+            )
 
 
+###############################################################################
+#                                                                             #
+#                           Combo Regularization                              #
+#                                                                             #
+###############################################################################
 
-    @Utils.timeIt
-    def _evalSmall(self, m):
-        r = self.Wsmall * ( self.mapping * (m - self.mref) )
-        return 0.5 * r.dot(r)
+class BaseComboRegularization(ObjectiveFunction.ComboObjectiveFunction):
 
-    @Utils.timeIt
-    def _evalSmallDeriv(self, m):
-        r = self.Wsmall * ( self.mapping * (m - self.mref) )
-        return r.T * ( self.Wsmall * self.mapping.deriv(m - self.mref) )
+    mapPair = Maps.IdentityMap
 
-    @Utils.timeIt
-    def _evalSmall2Deriv(self, m, v = None):
-        rDeriv = self.Wsmall * ( self.mapping.deriv(m - self.mref) )
-        if v is not None:
-            return rDeriv.T * (rDeriv * v)
-        return rDeriv.T * rDeriv
+    def __init__(
+        self, mesh, objfcts=[],
+        mapping=None, cell_weights=None, **kwargs
+        # mrefInSmooth=False, indActive=None,
+        # alpha_s=None, alpha_x=None, alpha_y=None, alpha_z=None,
+        # alpha_xx=None, alpha_yy=None, alpha_zz=None,
+    ):
 
-    @Utils.timeIt
-    def _evalSmoothx(self, m):
-        if self.mrefInSmooth == True:
-            r = self.Wx * ( self.mapping * (m - self.mref) )
-        elif self.mrefInSmooth == False:
-            r = self.Wx * ( self.mapping * (m) )
-        return 0.5 * r.dot(r)
+        super(BaseComboRegularization, self).__init__(
+            objfcts=[], multipliers=None
+        )
 
-    @Utils.timeIt
-    def _evalSmoothy(self, m):
-        if self.mrefInSmooth == True:
-            r = self.Wy * ( self.mapping * (m - self.mref) )
-        elif self.mrefInSmooth == False:
-            r = self.Wy * ( self.mapping * (m) )
-        return 0.5 * r.dot(r)
+        # self._cell_weights = cell_weights
+        self._mesh = mesh
+        self._mapping = mapping
+        self.objfcts = objfcts
 
-    @Utils.timeIt
-    def _evalSmoothz(self, m):
-        if self.mrefInSmooth == True:
-            r = self.Wz * ( self.mapping * (m - self.mref) )
-        elif self.mrefInSmooth == False:
-            r = self.Wz * ( self.mapping * (m) )
-        return 0.5 * r.dot(r)
+        Utils.setKwargs(self, **kwargs)
 
-    @Utils.timeIt
-    def _evalSmooth(self, m):
-        phiSmooth = self._evalSmoothx(m)
-        if self.regmesh.dim > 1:
-            phiSmooth += self._evalSmoothy(m)
-        if self.regmesh.dim > 2:
-            phiSmooth += self._evalSmoothz(m)
-        return phiSmooth
+    # Properties
+    alpha_s = Props.SimPEGFloat("smallness weight")
+    alpha_x = Props.SimPEGFloat("weight for the first x-derivative")
```

Comment:
Right now, all the combo regularizations use these.
`Tikhonov`, `Simple` and `Sparse` use them. 
We could instead move them to `Simple` and have `Tikhonov` and `Sparse` inherit `Simple`? (This choice is a bit arbitrary though). what do you think @rowanc1 ?